### ISSUE TITLE
how to make sure a Vec<u8> is serialized as `bytes`?

### ISSUE THREAD

**Issue by vmedea** [pyfisch/cbor#78]
I have a structure with a `Vec<u8>` that I serialize from Rust, using `cbor::ser::to_writer_packed`
```
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct LevelInfo {
    pub name: String,
    pub intro: Vec<Span>,
    pub code: Vec<u8>,
}
```
in Python, the `code` field deserializes to something like
```
 2: [127,
     69,
     76,
     70,
     1,
     1,
     1,
     0,
```
What I would expect is `b'\x7fELF\x01\0x01\0x01\0x00'`. It looks that it is using an array of individual values instead.
This is using a lot of extra space, is it possible to avoid this and use cbor's native bytes type?
(sorry if this is a stupid question, could not find anything about annotations!)

**Comment by sfackler** [comment]
You can use the [serde_bytes](https://crates.io/crates/serde_bytes) crate.

**Comment by medusacle** [comment]
awesome, thank you! basically just
```rust
    #[serde(with = "serde_bytes")]
    pub code: Vec<u8>,
```
and it worked it is embedded verbatim now

**Comment by vi** [comment]
Maybe information about `serde_bytes` should be more visible in main serde_cbor's README and root doccomment?

**Comment by vmedea** [comment]
Yes I agree it would be good to mention this interaction with serde_bytes, though looking through the manual I'm not sure where it would fit in. Maybe a section that specifies how rust types are mapped to CBOR types and vice versa would be useful, and could mention this.

**Comment by jawline** [comment]
If I migrate to serde_bytes is there any way to keep CBOR compatible with older versions of the document that didn't use byte strings, currently they are incompatible (I can an error that the deserializer expected a byte string)

**Comment by pyfisch** [comment]
@jawline You will need to write a custom deserializer that accepts either byte arrays or byte strings.

**Comment by jawline** [comment]
It seems I misunderstood my issue a little bit, serde bytes is actually compatible with the old encoded data for cbor but older clients are now incompatible with data produced serde bytes > cbor for Vec<u8>. This is a much more minor issue so everythings fine, thanks for replying!
